File: src/_1ProgrammingBasicsWithJavaMay2023/_4ForLoop/_2Exercise/_04CleverLily.java
package _1ProgrammingBasicsWithJavaMay2023._4ForLoop._2Exercise;

import java.util.Scanner;

public class _04CleverLily {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int age = Integer.parseInt(scanner.nextLine());
        double priceOfWashingMachine = Double.parseDouble(scanner.nextLine());
        int sellingPriceOfToy = Integer.parseInt(scanner.nextLine());
        int savedMoney = 0;
        int currentMoney = 10;
        int toy = 0;
        for (int i = 1; i <= age; i++) {
            if (i % 2 == 0) {
                savedMoney += currentMoney;
                currentMoney += 10;
                savedMoney -= 1;
            } else {
                toy += 1;
            }
        }
        int moneyFromToys = toy * sellingPriceOfToy;
        int totalSum = moneyFromToys + savedMoney;
        double diff = Math.abs(totalSum - priceOfWashingMachine);
        if (totalSum >= priceOfWashingMachine) {
            System.out.printf("Yes! %.2f", diff);
        } else {
            System.out.printf("No! %.2f", diff);
        }
    }
}

File: src/_1ProgrammingBasicsWithJavaScriptNovember2024/_4ForLoop/_2Exercise/_04CleverLily.js
function cleverLily(years, washingMachinePrice, sellingPricePerToy) {
    const LILI_BROTHER_STEAL_MONEY = 1;

    let numberOfToys = 0;
    let money = 0;

    for (let i = 1; i <= years; i++) {
        if (i % 2 === 0) {
            money += (i / 2) * 10;
            money -= LILI_BROTHER_STEAL_MONEY;
        } else {
            numberOfToys++;
        }
    }

    const moneyFromToys = numberOfToys * sellingPricePerToy;
    const totalMoney = money + moneyFromToys;

    const diff = Math.abs(totalMoney - washingMachinePrice);

    if (totalMoney >= washingMachinePrice) {
        console.log(`Yes! ${diff.toFixed(2)}`);
    } else {
        console.log(`No! ${diff.toFixed(2)}`);
    }
}